### ISSUE TITLE
Duplicate Refresh Tokens

### ISSUE THREAD

**Issue by jimgraham** [evonove/django-oauth-toolkit#377]
Hello

We tried to upgrade from 0.7.1 to 0.10.0 on the weekend. We started to see errors in our logs 

```
MultipleObjectsReturned: get() returned more than one RefreshToken -- it returned 2!
  File "django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "django/utils/decorators.py", line 34, in _wrapper
    return bound_func(*args, **kwargs)
  File "django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "django/utils/decorators.py", line 30, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "braces/views/_forms.py", line 22, in dispatch
    return super(CsrfExemptMixin, self).dispatch(*args, **kwargs)
  File "newrelic/hooks/framework_django.py", line 872, in wrapper
    return wrapped(*args, **kwargs)
  File "django/views/generic/base.py", line 89, in dispatch
    return handler(request, *args, **kwargs)
  File "django/utils/decorators.py", line 34, in _wrapper
    return bound_func(*args, **kwargs)
  File "django/views/decorators/debug.py", line 76, in sensitive_post_parameters_wrapper
    return view(request, *args, **kwargs)
  File "django/utils/decorators.py", line 30, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "oauth2_provider/views/base.py", line 170, in post
    url, headers, body, status = self.create_token_response(request)
  File "oauth2_provider/views/mixins.py", line 124, in create_token_response
    return core.create_token_response(request)
  File "oauth2_provider/oauth2_backends.py", line 138, in create_token_response
    headers, extra_credentials)
  File "oauthlib/oauth2/rfc6749/endpoints/base.py", line 64, in wrapper
    return f(endpoint, uri, *args, **kwargs)
  File "oauthlib/oauth2/rfc6749/endpoints/token.py", line 100, in create_token_response
    request, self.default_token_type)
  File "oauthlib/oauth2/rfc6749/grant_types/refresh_token.py", line 58, in create_token_response
    self.validate_token_request(request)
  File "oauthlib/oauth2/rfc6749/grant_types/refresh_token.py", line 101, in validate_token_request
    request.refresh_token, request.client, request):
  File "oauth2_provider/oauth2_validators.py", line 376, in validate_refresh_token
    rt = RefreshToken.objects.get(token=refresh_token)
```

In our db, there are 2 RefreshTokens with the same `token`. 

I can't really see how these get created. I assume there is a race condition somewhere in `save_bearer_token` https://github.com/evonove/django-oauth-toolkit/blob/master/oauth2_provider/oauth2_validators.py#L295 but I can't really see how it would happen.

Any thoughts on where to look?

**Comment by jleclanche** [comment]
Same issue as #429?

**Comment by kevin-brown** [comment]
This sounds like a duplicate of https://github.com/evonove/django-oauth-toolkit/issues/382, which was fixed in 0.11.0.
